src/routes/github/github-oauth-router.ts

```diff
@@ -8,17 +8,22 @@ import { Tracer } from "config/tracer";
 import { envVars } from "config/env";
 import { GithubAPI } from "config/github-api";
 import { Errors } from "config/errors";
-import { getGitHubHostname, getGitHubApiUrl } from "~/src/util/get-github-client-config";
+import { getGitHubHostname, getGitHubApiUrl, getGitHubClientConfigFromAppId } from "~/src/util/get-github-client-config";
 import { createHashWithSharedSecret } from "utils/encryption";
 
 const logger = getLogger("github-oauth");
-
-const githubClient = envVars.GITHUB_CLIENT_ID;
-const githubSecret = envVars.GITHUB_CLIENT_SECRET;
 const baseURL = envVars.APP_URL;
 const scopes = ["user", "repo"];
 const callbackPath = "/callback";
 
+const getRedirectUrl = async (req, res, state) => {
+	const { jiraHost, gitHubAppId } = res.locals;
+	const callbackURI = new URL(`${req.baseUrl + req.path}/..${callbackPath}`, baseURL).toString();
+	const gitHubHostname = await getGitHubHostname(jiraHost, gitHubAppId);
+	const { gitHubClientId } = await getGitHubClientConfigFromAppId(gitHubAppId, jiraHost);
```

Comment:
I think the clientId is already availble from the res.locals (provided from GitHubServerMiddlware). Maybe we can use that to save a db call?

Reply:
Pulled from `res.locals` instead of doing another db call.